phpstan.neon

```diff
@@ -10,4 +11,23 @@ parameters:
         - tests
     excludePaths:
         - tests/Fixtures
+    treatPhpDocTypesAsCertain: false
+    ignoreErrors:
+        - '/^Anonymous function should have native return typehint ".+"\.$/'
```

Comment:
> Using the baseline approach does not work as any contribution needs to update it anyway. So let's go back to a stable level and fix errors step by step as was done until level 5. Sorry about that.

You will never guess who is [the author](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4975#issue-423637598) of this 😄

I've tried (and seen others trying) using baseline file for PHPStan and the scenario was always the same:
- adding baseline (always over 1k lines, sometimes even 10k, we [had](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4975/files#diff-995edee38ad4f8387e58ebd52c31bcc04c56cc2448d331b1cf5e0b35c57b9efa) over 14k)
- regenerating after any bigger code changes ➡️ diff in PR was huge / complicated ➡️ ignored during review
- conclusion was made to switch from baseline to `ignoreErrors` RegExes

> These errors don't all look like we actually want to ignore them.

Definitely, but they are of different level - e.g. "Anonymous function should have native return typehint" would be very easy to remove, while for "should be contravariant with parameter" I'm not sure we will be even able to get rid of it.

Reply:
My comment back then was not completely exact 😛: what doesn't work is using the automatically generated baseline, which does not use regular expressions. I suppose it would work if we maintain it manually and use regular expressions. Though adding those here is fine too, but I would at least add some comment to separate these errors from the ones we can't fix at all, e.g. `# baseline`.

Reply:
how does this configuration prevent new cases if we don't use something like a base file, now looks to be me we add new strict tests and than ignore these again through configuration here or maybe I'm missing something?

Reply:
you are correct, this does not prevent new cases - same as it is right now on `master` - the idea is to merge this with few fixes and then remove each exception with fixes one-by-one - to avoid huge PRs, like this one: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5934 - which BTW should be renamed to "Add return types and many other thigs".

I made the review yesterday manually, finding few missing places - with this PR merged, one can simply remove the "Anonymous function should have native return typehint" exception and have much better feedback from PHPStan.

Reply:
The size of an PR is in itself not important, having a lot of small PR that keep creating conflicts over and over and over again to remove something like `@requires PHP7` or update an email adrress which both do no harm keeping what so ever, is what keep people spending hours fixing conflicts, building up bigger and bigger PR's. 

I'm :-1:  adding rules and than ignoring, these should be fixed or not no rule added, if that means it is a big PR than so be it. Btw. the PR should be called "Add rules and ignores them"

Reply:
> The size of an PR is in itself not important

it is, when changing 289 files in one PR it is much easier to miss a place like https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5934/files#diff-3ca0431e63da2df6c60a5b4c3f27edd1f977eee69b323c473b4bf805ff3efe1bR230

> having a lot of small PR that keep creating conflicts

I'd still prefer small PRs and conflicts in the spirit of SRP, than huge PRs hard to review.

> these should be fixed or not no rule added

Or made step-by-step, `phpstan/phpstan-strict-rules` exists for quite long time and no one added it yet as adding it without any exceptions is huge effort an I cannot imagine reviewing such PR, while I can for this and for fixing each exception separately.

Reply:
I see nothing wrong with linked code, so I fixed 100 things and the 101th thing, why don't you PR the fix if you care that much instate of lecturing me over fixing 100 things and not 101?

Reply:
Here it is: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5962 - I thought we aim in having small PRs for easier review, e.g. The Adventures of Julien and Braces Fixer, that's why I don't understand why sudden change of approach.